oozie-to-airflow/converter/oozie_converter.py

```diff
@@ -93,123 +92,76 @@ def __init__(
+
+        It uses the mapper, which is stored in ParsedNode. The result is saved in ParsedNode.tasks
+        and ParsedNode.relations
+        """
+        logging.info("Converting nodes to tasks and inner relations")
+        for p_node in nodes.values():
+            tasks, relations = p_node.mapper.to_tasks_and_relations()
+            p_node.tasks = tasks
+            p_node.relations = relations
 
     def add_properties_to_params(self, params: Dict[str, str]):
         """
         Template method, can be overridden.
         """
         return el_utils.parse_els(self.job_properties_file, params)
 
-    def create_dag_file(self, nodes: Dict[str, ParsedNode], depends: Set[str], relations: Set[Relation]):
+    def create_dag_file(self, workflow: Workflow):
         """
         Writes to a file the Apache Oozie parsed workflow in Airflow's DAG format.
-
-        :param nodes: A dictionary of {'task_id': ParsedNode object}
-        :param depends: A list of strings that will be interpreted as import
-            statements
-        :param relations: A list of Relation corresponding to operator relations
         """
         file_name = self.output_dag_name
         with open(file_name, "w") as file:
             logging.info(f"Saving to file: {file_name}")
-            self.write_dag(depends, file, nodes, relations)
+            dag_content = self.render_workflow(workflow)
+            file.write(dag_content)
         black.format_file_in_place(
             Path(file_name), mode=black.FileMode(line_length=110), fast=False, write_back=black.WriteBack.YES
         )
 
-    def write_dag(
-        self, depends: Set[str], file: TextIO, nodes: Dict[str, ParsedNode], relations: Set[Relation]
-    ):
+    def copy_extra_assets(self, nodes: Dict[str, ParsedNode]):
         """
-        Template method, can be overridden.
-        """
-        self.write_dependencies(file, depends)
-        self.write_params(file, self.params)
-        self.write_dag_header(file, self.dag_name, self.schedule_interval, self.start_days_ago)
-        self.write_nodes(file, nodes)
-        file.write("\n\n")
-        self.write_relations(file, relations)
-
-    @staticmethod
-    def write_params(file: TextIO, params: Dict[str, str]) -> None:
-        converted_params: Dict[str, Union[List[str], str]] = {
-            x: comma_separated_string_to_list(y) for x, y in params.items()
-        }
-        file.write("PARAMS = " + json.dumps(converted_params, indent=INDENT) + "\n\n")
-
-    def write_nodes(self, file: TextIO, nodes: Dict[str, ParsedNode], indent: int = INDENT):
-        """
-        Writes the Airflow tasks to the given opened file object.
-
-        :param file: The file pointer to write to.
-        :param nodes: Dictionary of {'task_id', ParsedNode}
-        :param indent: integer of how many spaces to indent entire operator
+        Copies additional assets needed to execute a workflow, eg Pig scripts
```

Comment:
```suggestion
        Copies additional assets needed to execute a workflow, eg. Pig scripts.
```

Reply:
I think we must have linter and grammar check indeed :). I will add issue for that :)